conda_lock/pypi_solver.py

```diff
@@ -278,12 +279,27 @@ def parse_pip_requirement(requirement: str) -> Optional[Dict[str, str]]:
     return match.groupdict()
 
 
+class PoetryDependencyWithHash(PoetryDependency):
+    def __init__(self, *args, hash: Optional[str] = None, **kwargs) -> None:
```

Comment:
I didn't include type hints for `*args` and `**kwargs`. I tried copying them from `PoetryDependency`, but that leads me to a dilemma:

With the copied type hints

```
class PoetryDependencyWithHash(PoetryDependency):
    def __init__(
        # ...
        extras=None,  # type: Union[List[str], FrozenSet[str]]
    )
```
mypy fails with the following error
```
Incompatible default for argument "extras" (default has type "None", argument has type "list[str] | frozenset[str]")  [assignment]
```

But when I make the `extras` argument optional,

```
class PoetryDependencyWithHash(PoetryDependency):
    def __init__(
        # ...
        extras=None,  # type: Optional[Union[List[str], FrozenSet[str]]]
    ) -> None:
        super().__init__(
            # ...
            extras=extras,
        )
```
there's a conflict when calling `super.__init__`
```
Argument "extras" to "__init__" of "Dependency" has incompatible type "list[str] | frozenset[str] | None"; expected "list[str] | frozenset[str]"
```

Do you have an idea how to solve this?

Reply:
The upstream type hint is wrong, so I think this would be a good case for `# type: ignore`.

Reply:
Ok, done!